client/tests/integration/smartcontracts/executor_with_custom_parameter/src/lib.rs

```diff
@@ -0,0 +1,68 @@
+//! Runtime Executor which defines a custom configuration parameter
+
+#![no_std]
+
+extern crate alloc;
+#[cfg(not(test))]
+extern crate panic_halt;
+
+use alloc::{format, string::String};
+
+use iroha_executor::{parameter::Parameter, prelude::*, DataModelBuilder};
+use iroha_schema::IntoSchema;
+use lol_alloc::{FreeListAllocator, LockedAllocator};
+use serde::{Deserialize, Serialize};
+
+#[global_allocator]
+static ALLOC: LockedAllocator<FreeListAllocator> = LockedAllocator::new(FreeListAllocator::new());
+
+getrandom::register_custom_getrandom!(iroha_executor::stub_getrandom);
+
+#[derive(Constructor, ValidateEntrypoints, Validate, Visit)]
+#[visit(custom(visit_register_domain, visit_set_parameter))]
+struct Executor {
+    verdict: Result,
+    block_height: u64,
+}
+
+fn visit_set_parameter(executor: &mut Executor, _authority: &AccountId, isi: &SetParameter) {
+    execute!(executor, isi);
+}
+
+fn visit_register_domain(executor: &mut Executor, _authority: &AccountId, isi: &Register<Domain>) {
+    // FIXME: unwrap is ok here?
+    let required_prefix = FindAllParameters
+        .execute()
+        .unwrap()
+        .into_iter()
+        .map(Result::unwrap)
```

Comment:
yes, because it would indicate a bug in iroha. It should be considered impossible to get Err here

Reply:
Got it, replaced with explanatory `.expect("...")`ations